app/experimenter/nimbus-ui/src/hooks/useCommonForm/index.test.tsx

```diff
@@ -144,5 +155,42 @@ describe("hooks/useCommonForm", () => {
         expect(screen.queryByTestId(name)).toBeInTheDocument();
       });
     });
+
+    it("FormBranch", () => {
+      const { container } = render(
+        <BranchSubject
+          experimentFeatureConfig={MOCK_FEATURE_CONFIG_WITH_SCHEMA}
+        />,
+      );
+
+      branchFieldNames.forEach((name) => {
+        const fieldName = `referenceBranch.${name}`;
+        // all branch field names call `formControlAttrs`
+        expect(screen.queryByTestId(fieldName)).toBeInTheDocument();
+
+        // featureValue must be "enabled" to show the submit error
+        if (name === "featureValue") {
+          (async () => {
+            const field = container.querySelector(
+              `[name="${fieldName}"]`,
+            ) as HTMLInputElement;
+            act(() => {
+              fireEvent.focus(field!);
+              fireEvent.change(field!, { target: { value: true } });
+            });
+            await waitFor(() => {
+              expect(
+                screen.queryByTestId(`${fieldName}-form-errors`),
+              ).toBeInTheDocument();
+            });
+          })();
+          // featureEnabled does not call FormErrors because it is a global error
+        } else if (name !== "featureEnabled") {
```

Comment:
Yeah, like I guess this bit... `featureEnabled` doesn't really have anything specifically to do with useCommonForm itself since it's a local concern of the branch forms. Still... we can worry about this when useCommonNestedForm is applied to a wider variety. (or not worry about it at all if it's not worth it practically speaking)

Reply:
Yeah, I pinged with Jody a bit in Slack about this, I'm cool with moving them to their form files. When I made this test file, my intent was to mock what `useCommonForm` returns to check that it'd been called with `name`, then had trouble getting it working and filed EXP-780 / #4312.

I think I'll add it as a comment in #4312 to look into moving the tests out.

Reply:
OK, adjusted the TODO comment in the code and added a comment in the ticket as well: https://github.com/mozilla/experimenter/issues/4312#issuecomment-764990333